### DISCUSSION TITLE
pg-promise equivalent to libpq-async PQsendQuery?

### DISCUSSION THREAD

**Discussion by dennyau** [vitaly-t/pg-promise#885] [Q&A] [answered]
Hi everyone, first time poster - long time lurker.

I want to submit to my postgres db a long running query that generates a new table for reading later.
I do not need any returned information/errors when I submit the query.

According to PG docs, a command can be submitted to the server without waiting for the result via `PQsendQuery()`
https://www.postgresql.org/docs/current/libpq-async.html

Does pg-promise support this?
I have experience with properly chaining the pg-promise blocks to expect results.  However I cannot find proper documentation that pg-promise without async-await is actually "fire-and-forget".
For example, I've seen errors where a long running query with pg-promise will timeout, have the connection be dropped and the entire transaction rolled back on the database server.

In summary, can I use pg-promise to:
1) submit a long running query to the db that does not expect a return
2) close the connection safely, without causing the long running query to roll back?

Thanks!

**Comment by vitaly-t** [comment, marked as answer]
This library supports only what [the underlying driver](https://github.com/brianc/node-postgres) does.

To my knowledge, there is no such thing as async execution, as the driver requires that only one query can be executed at a time against the same connection, so you would have to wait. You can, of course, execute a query and simply not wait for the result, but that's different, because you still won't be able to execute another one, till it finishes.

There have been some discussions around it in the driver:

* https://github.com/brianc/node-postgres/issues/660
* https://github.com/brianc/node-postgres/issues/2646
